src/backend/db/database_manager.py

```diff
@@ -418,5 +420,421 @@ def delete_custom_field(self, customer_guid, field_name):
+                        INSERT INTO custom_field_values (ticket_id, {columns}) 
+                        VALUES (:ticket_id, {placeholders})
+                    '''
+                    params = {"ticket_id": ticket_id}
+                    params.update(custom_fields)
+
+                    logger.debug(f"Inserting custom fields: {custom_fields}")
+                    session.execute(text(query_for_insert_custom_fields), params)
+
+            session.commit()
+            logger.debug(f"Ticket ID {ticket_id} created successfully with custom fields")
+
+            return {"ticket_id": ticket_id, "status": "created"}
+
+        except ValueError as e:
+            logger.error(f"Validation error: {e}")
+            session.rollback()
+            raise e  # Propagate the ValueError to the calling function
+
+        except SQLAlchemyError as e:
+            logger.error(f"Database error creating ticket for customer {customer_guid}: {e}")
+            session.rollback()
+            raise e  # Propagate the SQLAlchemyError to the calling function
+        except OperationalError as e:
+            logger.error(f"Database connectivity issue: {e}")
+            raise e
+        except DatabaseError as e:
+            logger.error(f"Database connectivity issue: {e}")
+            raise e
+        except Exception as e:
+            logger.error(f"Unexpected error while creating ticket: {e}")
+            session.rollback()
+            raise e  # Propagate the general exception to the calling function
+
+        finally:
+            session.close()
+
+    def get_ticket_by_id(self, ticket_id, customer_guid):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+        try:
+
+            # Check if the database exists
+            logger.debug(f"Checking if database {customer_db_name} exists")
+            result = session.execute(
+                text("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = :db_name"),
+                {"db_name": customer_db_name}
+            ).fetchone()
+
+            if not result:
+                raise ValueError(f"Database {customer_db_name} does not exist")
+
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Retrieve ticket base details
+            ticket_query = '''
+                SELECT ticket_id, chat_id, title, description, priority, status
+                FROM tickets
+                WHERE ticket_id = :ticket_id
+            '''
+            ticket_result = session.execute(
+                text(ticket_query),
+                {"ticket_id": ticket_id},
+            ).fetchone()
+
+            if not ticket_result:
+                return None  # If ticket is not found, return None
+
+            # Process ticket base data
+            ticket_data = {
```

Comment:
use

ticket_result.keys() to construct the JSON instead of hard coding numbers

ticket_result[0] ticket_result[1] etc.,

Reply:
Modified sir